src/borg/repository.py

```diff
@@ -138,6 +139,7 @@ def __enter__(self):
         if self.do_create:
             self.do_create = False
             self.create(self.path)
+            self.created = True
```

Comment:
Well do_create tells us that we need to create it on the first open. created tells us that the repository was created in the current session. You can't leave do_create on True, because then reopen breaks.